torch/distributed/_spmd/distribute.py

```diff
@@ -491,7 +549,17 @@ def _convert_to_distributed(
                 # prevent running this collective in backwards pass
                 run_check=False,
             )
-
+        elif node.target == aten.sym_size:
+            dim: int = node.args[1]
+            dtensor: DTensor = cast(DTensor, node_to_obj[node.args[0]])
+            placement: Placement = (
+                Shard(dim)
+                if any(p.is_shard(dim) for p in dtensor.placements)
+                else Replicate()
+            )
+            node_to_obj[node] = DSize(
+                size=dtensor._local_tensor.size(dim), placement=placement
```

Comment:
nit: use `to_local` instead?

Reply:
I saw that calls into an Autograd function + a detach, so wanted to avoid those initially. But yep, let us use the public API from DTensor